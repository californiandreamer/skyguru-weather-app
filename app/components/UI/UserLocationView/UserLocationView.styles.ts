import { littleGap, smallGap } from 'app/constants/values'
import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    paddingVertical: smallGap,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  locationIcon: {
    paddingRight: littleGap,
  },
})
