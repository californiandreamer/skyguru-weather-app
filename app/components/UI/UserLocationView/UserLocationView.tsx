import { PinIcon } from 'app/assets'
import StyledText from 'app/components/HOC/Text'
import {
  defaultIconSize,
  pressOpacity,
  smallFontSize,
} from 'app/constants/values'
import { showSearch } from 'app/store/actions/search'
import { RootState } from 'app/store/reducers/rootReducer'
import React from 'react'
import { TouchableOpacity, View } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'

import styles from './UserLocationView.styles'

const UserLocationView: React.FC = () => {
  const dispatch = useDispatch()

  const { pending, currentWeather, error } = useSelector(
    (state: RootState) => state.currentWeather
  )

  const city = currentWeather?.name

  return (
    <TouchableOpacity
      style={styles.container}
      activeOpacity={pressOpacity}
      onPress={() => dispatch(showSearch())}
    >
      {!pending && !error && city ? (
        <>
          <View style={styles.locationIcon}>
            <PinIcon width={defaultIconSize} height={defaultIconSize} />
          </View>
          <StyledText fontSize={smallFontSize}>{city}</StyledText>
        </>
      ) : null}
    </TouchableOpacity>
  )
}

export default UserLocationView
