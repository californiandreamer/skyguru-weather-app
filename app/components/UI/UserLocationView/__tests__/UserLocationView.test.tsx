import configureStore from 'app/store'
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import UserLocationView from '../UserLocationView'

describe('UserLocationView component', () => {
  it('renders correctly', () => {
    const store = configureStore()

    const tree = renderer
      .create(
        <Provider store={store}>
          <UserLocationView />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
