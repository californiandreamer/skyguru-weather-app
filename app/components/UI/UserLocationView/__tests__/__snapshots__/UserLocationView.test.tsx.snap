// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLocationView component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
/>
`;
